qiskit_experiments/characterization/ef_spectroscopy.py

```diff
@@ -0,0 +1,44 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Spectroscopy for the e-f transition."""
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Gate
+
+from qiskit_experiments.characterization.qubit_spectroscopy import QubitSpectroscopy
+
+
+class EFSpectroscopy(QubitSpectroscopy):
+    """Class that runs spectroscopy on the e-f transition by scanning the frequency.
+
+    The circuits produced by spectroscopy, i.e.
+
+    .. parsed-literal::
+
+                   ┌───┐┌────────────┐ ░ ┌─┐
+              q_0: ┤ X ├┤ Spec(freq) ├─░─┤M├
+                   └───┘└────────────┘ ░ └╥┘
+        measure: 1/═══════════════════════╩═
+                                          0
+
+    """
+
+    @staticmethod
+    def _template_circuit(freq_param) -> QuantumCircuit:
+        """Return the template quantum circuit."""
+        circuit = QuantumCircuit(1)
+        circuit.x(0)
```

Comment:
Do you think this is sufficiently general to cover more higher level excitation (I think this is what you have in your mind)? This seems like we need to add custom x12 gate definition to run 2-3 spectroscopy, along with the schedule of 2-3 spect pulse. So we need mechanism that passes multiple custom gate schedules to corresponding custom gates.

Reply:
No this will not cover higher levels (e.g. 2-3) which will require additional spectroscopy classes but it conforms to your comment above: https://github.com/Qiskit/qiskit-experiments/pull/82#pullrequestreview-677194650. That being said with this refactor new spectroscopy experiments do not require too much additional code.

Reply:
I understand. I think ge + ef cals are sufficient in most cases and this implementation is sufficient for us. These higher level spectroscopy experiments can be supported by third party library, or we could write separate experiment package in future.